lib/compile.js

```diff
@@ -23,10 +23,12 @@ module.exports = {
         if (stats.compilation.errors.length) {
           throw new Error('Webpack compilation error, see above');
         }
-        const outputPath = stats.compilation.compiler.outputPath;
-        this.webpackOutputPath = outputPath;
-        this.originalServicePath = this.serverless.config.servicePath;
-        this.serverless.config.servicePath = outputPath;
+        if (!this.options.location) {
+          const outputPath = stats.compilation.compiler.outputPath;
+          this.webpackOutputPath = outputPath;
+          this.originalServicePath = this.serverless.config.servicePath;
+          this.serverless.config.servicePath = outputPath;
+        }
```

Comment:
I'm not sure if changing the serverless servicePath here will have side-effects on all other functions. The problem is, that Serverless uses this path for it's package command, persists it's service state there, which it reloads when entering the deploy phase. This change does not only affect the offline use case, but also the package/deploy use cases.

Other plugins rely on the service path to be set as Serverless sets it.

I's propose to remove this change from the PR (it is imo not necessary to fix the race condition).